packages/suite-base/src/components/PanelSettings/index.tsx

```diff
@@ -134,17 +133,32 @@ export default function PanelSettings({
 
   const [config, , extensionSettings] = useConfigById(selectedPanelId);
   const messagePipelineState = useMessagePipelineGetter();
-  const topicToSchemaNameMap = getTopicToSchemaNameMap(messagePipelineState());
 
-  const settingsTree = usePanelStateStore((state) =>
-    buildSettingsTree({
+  const storedSettingsTrees = usePanelStateStore(({ settingsTrees }) => settingsTrees);
+  const settingsTree = useMemo(
+    () =>
+      buildSettingsTree({
+        config,
+        extensionSettings,
+        messagePipelineState,
+        panelType,
+        selectedPanelId,
+        settingsTrees: storedSettingsTrees,
+      }),
+    [
       config,
       extensionSettings,
+      messagePipelineState,
       panelType,
       selectedPanelId,
-      state,
-      topicToSchemaNameMap,
-    }),
+      /**
+       * The core issue is that settingsTrees object in the PanelStateStore is being
+       * mutated on each render, leading to unnecessary calls to buildSettingsTree
+       * To address this, we need to ensure that settingsTrees remains
+       * referentially stable unless its actual content changes.
+       */
+      storedSettingsTrees,
```

Comment:
I tried an alternative to this, to try render only when needed but indeed, that way seems to be the most effective way for now, without generating side effects. I would say it was a great finding. Thanks for this @luluiz !